tidb-binlog/pump_client/client.go

```diff
@@ -153,19 +150,22 @@ func NewPumpsClient(etcdURLs string, timeout time.Duration, securityOpt pd.Secur
 		EtcdRegistry:       node.NewEtcdRegistry(cli, DefaultEtcdTimeout),
 		Pumps:              NewPumpInfos(),
 		Selector:           NewSelector(Range),
-		RetryTime:          DefaultAllRetryTime,
 		BinlogWriteTimeout: timeout,
 		Security:           security,
 	}
 
-	err = newPumpsClient.getPumpStatus(ctx)
+	revision, err := newPumpsClient.getPumpStatus(ctx)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+
+	if len(newPumpsClient.Pumps.Pumps) == 0 {
```

Comment:
it's hard to modify the name now, because tools relay on tidb, and tidb relay on tools too, this variable is used in tidb's unit test, if change this name, will build failed.

Reply:
if it's not right, we can spend time to fix it. I suggest to create an issue and fix it later

Reply:
ok, I will create a issue for it